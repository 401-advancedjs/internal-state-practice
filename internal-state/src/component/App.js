import React from 'react';
import Form from './Form.js';

export default class App extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Form />
      </React.Fragment>
    );
  }
}