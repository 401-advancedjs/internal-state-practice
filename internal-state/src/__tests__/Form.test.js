import React from 'react';
import Form from '../component/Form.js';
import App from '../component/App.js';
import renderer from 'react-test-renderer';

describe('<Form />', () => {
  test('basic rendering', () => {
    const mountedForm = shallow(<Form />);
    console.log(mountedForm);
    expect(mountedForm.find('form')).toBeTruthy();
  });

  test('form input change', () => {
    const mountedApp = mount(<App />);
    const name = mountedApp.find('input.name');
    name.simulate('change', {target: {value: 'Melissa'}});
    expect(mountedApp.state('name')).toEqual('Melissa');

  });
  
  test('increment clicks', () => {
    const mountedApp = mount(<App />);
    const submit = mountedApp.find('input.submitButton');
    submit.simulate('click');
    expect(mountedApp.state('count')).toBe(1);;
  });

  test('rendering follows the snapshot', () => {
    const snapshot = renderer.create(<Form />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});