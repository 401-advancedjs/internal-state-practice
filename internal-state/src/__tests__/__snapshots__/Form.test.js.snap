// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> rendering follows the snapshot 1`] = `
<form>
  <label>
    Name:
    <input
      name=""
      onChange={[Function]}
      type="text"
    />
  </label>
  <div>
    # of Updates: 
    0
  </div>
  <div>
    This is your name: 
    
  </div>
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</form>
`;
